source/val/validate_ray_tracing.cpp

```diff
@@ -0,0 +1,199 @@
+        return _.diag(SPV_ERROR_INVALID_DATA, inst)
+               << "Ray Origin must be a 32-bit float 3-component vector";
+      }
+
+      const uint32_t ray_tmin = _.GetOperandTypeId(inst, 7);
+      if (!_.IsFloatScalarType(ray_tmin) || _.GetBitWidth(ray_tmin) != 32) {
+        return _.diag(SPV_ERROR_INVALID_DATA, inst)
+               << "Ray TMin must be a 32-bit float scalar";
+      }
+
+      const uint32_t ray_direction = _.GetOperandTypeId(inst, 8);
+      if (!_.IsFloatVectorType(ray_direction) ||
+          _.GetDimension(ray_direction) != 3 ||
+          _.GetBitWidth(ray_direction) != 32) {
+        return _.diag(SPV_ERROR_INVALID_DATA, inst)
+               << "Ray Direction must be a 32-bit float 3-component vector";
+      }
+
+      const uint32_t ray_tmax = _.GetOperandTypeId(inst, 9);
+      if (!_.IsFloatScalarType(ray_tmax) || _.GetBitWidth(ray_tmax) != 32) {
+        return _.diag(SPV_ERROR_INVALID_DATA, inst)
+               << "Ray TMax must be a 32-bit float scalar";
+      }
+
+      const Instruction* payload = _.FindDef(inst->GetOperandAs<uint32_t>(10));
+      if (payload->opcode() != SpvOpVariable) {
+        return _.diag(SPV_ERROR_INVALID_DATA, inst)
+               << "Payload must be the result of a OpVariable";
+      } else if (payload->word(3) != SpvStorageClassRayPayloadKHR &&
+                 payload->word(3) != SpvStorageClassIncomingRayPayloadKHR) {
+        return _.diag(SPV_ERROR_INVALID_DATA, inst)
+               << "Payload must have storage class RayPayloadKHR or "
+                  "IncomingRayPayloadKHR";
+      }
+      break;
+    }
+
+    case SpvOpReportIntersectionKHR: {
+      _.function(inst->function()->id())
+          ->RegisterExecutionModelLimitation(
+              [](SpvExecutionModel model, std::string* message) {
+                if (model != SpvExecutionModelIntersectionKHR) {
+                  if (message) {
+                    *message =
+                        "OpReportIntersectionKHR requires IntersectionKHR "
+                        "execution model";
+                  }
+                  return false;
+                }
+                return true;
+              });
+
+      if (!_.IsBoolScalarType(result_type)) {
+        return _.diag(SPV_ERROR_INVALID_DATA, inst)
+               << "expected Result Type to be bool scalar type";
+      }
+
+      const uint32_t hit = _.GetOperandTypeId(inst, 2);
+      if (!_.IsFloatScalarType(hit) || _.GetBitWidth(hit) != 32) {
+        return _.diag(SPV_ERROR_INVALID_DATA, inst)
+               << "Hit must be a 32-bit int scalar";
+      }
+
+      const uint32_t hit_kind = _.GetOperandTypeId(inst, 3);
+      if (!_.IsUnsignedIntScalarType(hit_kind) ||
+          _.GetBitWidth(hit_kind) != 32) {
+        return _.diag(SPV_ERROR_INVALID_DATA, inst)
+               << "Hit Kind must be a 32-bit unsigned int scalar";
+      }
+      break;
+    }
+
+    case SpvOpExecuteCallableKHR: {
```

Comment:
SBT index? Spec doesn't really say much, but certainly it should be an integer. Not sure if it is restricted to 32 bits though.

Reply:
Ya, I noticed that already and made this with the PR
https://github.com/KhronosGroup/SPIRV-Registry/issues/156

I was going to wait for the WG to just confirm/respond and then add it in `spirv-val`

Reply:
That seems reasonable.

Reply:
https://github.com/KhronosGroup/SPIRV-Registry/pull/160 - will make new PR to add it